m4/README

```diff
@@ -0,0 +1,4 @@
+m4 macros needed by automake system.
+
+ax_require_defined.m4: needed by ax_cxx_compile_stdcxx.m4
+
```

Comment:
Should there be more files from the [Autoconf Archive](https://www.gnu.org/software/autoconf-archive/) listed here?